subsys/net/ip/net_context.c

```diff
@@ -816,6 +816,24 @@ NET_CONN_CB(tcp_established)
 				     sys_get_be32(NET_TCP_HDR(pkt)->ack));
 	}
 
+//printk("%u %u\n", sys_get_be32(NET_TCP_HDR(pkt)->seq), context->tcp->send_ack);
+
+	/* Peer sent us packet we've already seen. Apparently, our ack was lost. */
+	if (sys_get_be32(NET_TCP_HDR(pkt)->seq) < context->tcp->send_ack) {
+printk("Received already seen pkt - ACKing\n");
+
+	set_appdata_values(pkt, IPPROTO_TCP);
```

Comment:
> And yep, we can't do this (if taken literally) - we need to ACK what peer asked us for, not what we currently expect from it.

I thought that sending ACK to some value would automatically acknowledge all the previous values. RFC 793 in chapter 3.3.  Sequence Numbers says "The acknowledgment mechanism employed is cumulative so that an acknowledgment of sequence number X indicates that all octets up to but not including X have been received." https://tools.ietf.org/html/rfc793#section-3.3